src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -43,6 +43,8 @@ export interface QueryEditorProps {
   isInvalid?: boolean;
   queryEditorHeaderRef: React.RefObject<HTMLDivElement>;
   queryEditorHeaderClassName?: string;
```

Comment:
sorry about that.

i might have ported this from your changes. did you prefer to use the header ref?\ I don't think we are using it for anything else.

Reply:
i just picked my previous PRs and added 8b0af9d. what did you port that can replace this?

Reply:
i copied over this entire file and then cleaned up stuff that wasn't available or from other people's stuff but i accidentally left this in from your stuff.

Reply:
oh i see, git didn't complain since i was picking the same modification. but the extension uses the header ref and banner ref to mount, current code is intended

the banner one was added after previous PRs, because UX requires some callout above language selector